src/lib/rico-angular.service.ts

```diff
@@ -53,10 +57,41 @@ export class RicoService {
 
   createController(name: string): Promise<ControllerProxy> {
     if (this.clientContext && this.clientContext.isConnected) {
-      const controllerProxy = new ControllerProxy(this.clientContext);
-      return controllerProxy.create(name);
+      return new Promise<ControllerProxy>((resolve, reject) => {
+        const controllerProxy = new ControllerProxy(name, this.clientContext);
+        this.addToQueue(controllerProxy, resolve, reject);
+      });
     } else {
       return Promise.reject('Cannot create controller. ClientContext not conntected');
     }
   }
+
+  private addToQueue(controllerProxy: ControllerProxy, resolve: Function, reject: Function) {
+    const promise = new Promise((resolve, reject) => {
+      this.queue.push(resolve);
+    });
+
+    promise.then(() => {
+      controllerProxy.create().then((controllerProxy) => {
```

Comment:
[tslint] Shadowed name: 'controllerProxy' [no-shadowed-variable]
(same for reject)